src/index.js

```diff
@@ -27,8 +27,19 @@ function PaymentsClient(config) {
     throw new Error('paymentHost must be http or https');
   }
 
-  if (typeof this.product !== 'string') {
-    throw new Error('A \'product\' string must be provided');
+  if (typeof config.product !== 'object' || !config.product.id) {
+    throw new Error('A product id must be provided in a product object');
+  }
+
+  var productImageUrl = config.product.image;
+  if (!productImageUrl) {
+    console.warn(
+      'A product image URL was not supplied as a property of config.product.');
+  } else {
+    var imageProtocol = utils.getProtocol(productImageUrl);
+    if (imageProtocol !== 'http:' && imageProtocol !== 'https:') {
+      throw new Error('product.image must be served over http/https');
```

Comment:
The httpsOnly currently relates more to where our content is hosted. But I guess it should also apply to window.location.protocol as well as the image setting too.